interfaces/agnostic/watched_lit_succ_gen.cxx

```diff
@@ -0,0 +1,47 @@
+#include <watched_lit_succ_gen.hxx>
+#include <strips_prob.hxx>
+#include <strips_state.hxx>
+#include <action.hxx>
+
+namespace aptk {
+
+void WatchedLitSuccGen::init(){
+	watchers.resize(prob.num_fluents());
+	for(unsigned op = 0; op < prob.num_actions(); ++op){
+		auto act = prob.actions()[op];
+		auto& precs = act->prec_vec();
+		unsigned f = precs[0];
+		for(unsigned i = 1; i < precs.size(); ++i){
+			if(watchers[precs[i]].size() < watchers[f].size())
+				f = precs[i];
+		}
+		watchers[f].push_back(act->index());
+	}
+}
+
+inline bool WatchedLitSuccGen::iterator::applicable(){
```

Comment:
I can't define in the header because they depend on the definition of the fwd_search_prob, which in turn relies on the successor generators, so I get forward declaration errors...